controllers/drplacementcontrol_controller.go

```diff
@@ -470,8 +521,12 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		}
 	}
 
-	// delete cloned placementrule
-	return r.deleteClonedPlacementRule(ctx, clonedPlRuleName, drpc.Namespace)
+	// delete cloned placementrule, if created
```

Comment:
@BenamarMk here and at 502 is another fix for finalize time issues, when user sets the preferredCluster. (as we no longer create cloned Pl if user sets the preferredCluster)